test/unit/componentOverrides.test.tsx

```diff
@@ -112,3 +128,125 @@ describe("Theme component overrides", () => {
         });
     });
 });
+
+const MockSession = {
+    addEventListener: jest.fn(),
+    getUserId: jest.fn(),
+    getAccessTokenPayloadSecurely: jest.fn(),
+    doesSessionExist: jest.fn(),
+    validateClaims: jest.fn(),
+    validateGlobalClaimsAndHandleSuccessRedirection: jest.fn(),
+};
+
+const setMockResolvesSession = (ctx: SessionContextType) => {
+    if (ctx.loading === true) {
+        // We "simulate" loading by returning these promises that won't ever resolve
+        MockSession.getUserId.mockReturnValue(new Promise<any>(() => {}));
+        MockSession.getAccessTokenPayloadSecurely.mockReturnValue(new Promise<any>(() => {}));
+        MockSession.doesSessionExist.mockReturnValue(new Promise<any>(() => {}));
+        MockSession.validateClaims.mockReturnValue(new Promise<any>(() => {}));
+    } else {
+        MockSession.getUserId.mockResolvedValue(ctx.userId);
+        MockSession.getAccessTokenPayloadSecurely.mockResolvedValue(ctx.accessTokenPayload);
+        MockSession.doesSessionExist.mockResolvedValue(ctx.doesSessionExist);
+        MockSession.validateClaims.mockReturnValue(ctx.invalidClaims);
+    }
+};
+
+jest.spyOn(Session, "getInstanceOrThrow").mockImplementation(() => MockSession as any);
+
+describe("Components override per recipe provider", () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+        EmailPassword.reset();
+        SuperTokens.reset();
+
+        SuperTokens.init({
+            appInfo: {
+                apiBasePath: "/auth",
+                apiDomain: "http://localhost:3001",
+                appName: "JestTest",
+                websiteBasePath: "/auth",
+                websiteDomain: "http://localhost:3000",
+            },
+            recipeList: [
+                ThirdPartyEmailPassword.init({
+                    signInAndUpFeature: {
+                        providers: [Github.init()],
+                    },
+                    useShadowDom: false,
+                }),
+                EmailPassword.init({
+                    useShadowDom: false,
+                }),
+            ],
+        });
+        setMockResolvesSession({
+            userId: "mock-user-id",
+            accessTokenPayload: {},
+            invalidClaims: [],
+            doesSessionExist: false,
+            loading: false,
+        });
+    });
+
+    it.skip("Should not affect other recipes components when the same component is overridden", async () => {
```

Comment:
Why is this skipped?

Reply:
Please add another version of this test without `ThirdpartyEmailPasswordComponentsOverrideProvider` and check that the default is displayed.

Reply:
Actually, the other test-case already includes checking with two different recipe specific contexts, so you can just update this test case